src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsService.java

```diff
@@ -361,64 +362,24 @@ public boolean buildPlanIdIsValid(String projectKey, String buildPlanId) {
         }
     }
 
-    @Override
-    public Optional<Result> retrieveLatestBuildResult(ProgrammingExerciseParticipation participation, ProgrammingSubmission submission) {
-        final var report = fetchLatestBuildResultFromJenkins(participation);
-
-        // The retrieved build result must match the commitHash of the provided submission.
-        if (report.getCommits().stream().map(CommitDTO::getHash).noneMatch(hash -> hash.equals(submission.getCommitHash()))) {
-            return Optional.empty();
-        }
-
-        final var result = createResultFromBuildResult(report, (Participation) participation);
-        result.setRatedIfNotExceeded(report.getRunDate(), submission);
-        result.setSubmission(submission);
-
-        submission.setBuildFailed(result.getResultString().equals("No tests found"));
-        programmingSubmissionRepository.save(submission);
-
-        return Optional.empty();
-    }
-
     private void addFeedbackToResult(Result result, TestResultsDTO report) {
         // No feedback for build errors
         if (report.getResults() == null || report.getResults().isEmpty()) {
             result.setHasFeedback(false);
             return;
         }
 
-        final var feedbacks = report.getResults().stream().flatMap(testsuite -> testsuite.getTestCases().stream()).map(testCase -> {
-            final var feedback = new Feedback();
-            feedback.setPositive(testCase.getErrors() == null && testCase.getFailures() == null);
-            feedback.setText(testCase.getName());
-            String errorMessage = null;
-            // If we have errors or failures, they will always be of length == 1 since JUnit (and the format itself)
-            // should generally only report the first failure in a test case
-            if (testCase.getErrors() != null) {
-                errorMessage = testCase.getErrors().get(0).getMessage();
-            }
-            else if (testCase.getFailures() != null) {
-                errorMessage = testCase.getFailures().get(0).getMessage();
-            }
-            // The assertion message can be longer than the allowed char limit, so we shorten it here if needed.
-            if (errorMessage != null && errorMessage.length() > FEEDBACK_DETAIL_TEXT_MAX_CHARACTERS) {
-                errorMessage = errorMessage.substring(0, FEEDBACK_DETAIL_TEXT_MAX_CHARACTERS);
-            }
-            feedback.setDetailText(errorMessage);
+        final ProgrammingLanguage programmingLanguage = ((ProgrammingExercise) result.getParticipation().getExercise()).getProgrammingLanguage();
 
-            return feedback;
-        }).collect(Collectors.toList());
+        report.getResults().forEach(testsuite -> testsuite.getTestCases().forEach(testCase -> {
```

Comment:
Is there a reason why we use two nested `forEach` calls here instead of the `flatMap` and `collect` that was there before?

Reply:
Because of the result.addFeedback() method we don't need to .stream() anymore. I did this to be consistent with the BambooService

Reply:
Changed it to normal for each loops.